src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.ts

```diff
@@ -62,38 +73,22 @@ export class GbVariablesComponent implements OnInit {
     }
   }
 
-  private listAvailableViewModes(): SelectItem[] {
-    return Object.keys(VariablesViewMode).map(c => {
-      return {
-        label: VariablesViewMode[c],
-        value: VariablesViewMode[c]}
-    }) as SelectItem[];
-  }
-
-  get viewMode(): VariablesViewMode {
-    return this.variableService.variablesViewMode;
-  }
-
-  set viewMode(value: VariablesViewMode) {
-    this.variableService.variablesViewMode = value;
+  get allChecked(): boolean {
+    this._allChecked = this.variableService.variables.every((v: Concept) => {
```

Comment:
What about having this iteration inside the service and call `this.variableService.allChecked()` instead?

Reply:
yes, this is debatable, my idea is to only put things inside services when it is really necessary, otherwise it is really easy to blow up services and make them unmanageable. in this case, the allChecked flag is just something that is required by the UI component to render a checkbox, it is not used anywhere else, so it is sth. more belonging to the component. The services are for storing shared variables and necessary fetching and  transformation logics on those variables.